src-ocaml/tea_debug.ml

```diff
@@ -14,262 +14,304 @@ type 'model debug_model = {
   show_details : bool;
 }
 
-let debug :
-  ('msg -> string) ->
-  ('flags, 'model, 'msg) Tea_app.program ->
-  ('flags, 'model debug_model, 'msg debug_msg) Tea_app.program
-  =
-  let client_msg msg = ClientMsg msg in
-  fun string_of_msg { init; update; view; subscriptions; shutdown } ->
-    let init' (flags : 'flags) =
-      let cmodel, cmd = init flags in
-      {
-        history = ["_init_", cmodel];
-        state = Running;
-        show_details = false;
-      }, cmd |> Tea_cmd.map client_msg
-    in
+let debug string_of_msg update view subscriptions shutdown =
```

Comment:
Any chance of adding types to this function again, when they get complex enough I like to add them for documentative purposes (and to help ensure that I don't break it in the future ^.^; ).  :-)

Reply:
Done.